src/WebJobs.Script/Workers/Rpc/IWorkerConfigurationResolver.cs

```diff
@@ -13,8 +11,8 @@ namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc
     internal interface IWorkerConfigurationResolver
     {
         /// <summary>
-        /// Retrieves the worker configuration resolution information which includes the root directory path of workers and worker configuration paths.
+        /// Retrieves a dictionary of worker configurations, keyed by language name.
         /// </summary>
-        WorkerConfigurationInfo GetConfigurationInfo();
+        Dictionary<string, RpcWorkerConfig> GetWorkerConfigs();
```

Comment:
Consider returning `IReadOnlyDictionary`, unless you expect callers to mutate it.

Preferring `IReadOnlyDictionary` opens up the door for implementations to cache their result and return it in future calls, as they can assert no caller has mutated it underneath them

Reply:
Caller can modify it - [Link1](https://github.com/Azure/azure-functions-host/blob/febfc3c66e5c21fdb1dbfcae07f02ff382fbef93/src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs#L54), [Link2](https://github.com/Azure/azure-functions-host/blob/febfc3c66e5c21fdb1dbfcae07f02ff382fbef93/src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs#L66)

Reply:
If we are modifying after resolving, maybe an aggregator pattern will be more appropriate so a separate provider can append those extra configs.